codefori/vscode-db2i
# Implement syntax checking features

Introduce a problem provider that highlights SQL syntax issues and allows for configurable syntax check intervals. This feature will only syntax check as you are typing. If you're working in a document made up of multiple statements, it will only check the statement you're currently working on while you type. You can use the newly added button to check an entire document.

How to test while editing:

1. Connect to IBM i with Code for IBM i
2. Open an SQL document (new or existing)
3. Start typing. There is wait period before the syntax is checked (configurable), but the default is 1.5 seconds.

Test the _Check Syntax_ button:

1. Connect to IBM i with Code for IBM i
2. Open an SQL document (new or existing)
3. Press the _Check Syntax_ button, which appears next to the Run button in the editor bar in the top right.

<img width="319" alt="image" src="https://github.com/user-attachments/assets/017109bf-eb4e-41e2-ae58-701b0f5c5e62" />
